hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsInputStream.java

```diff
@@ -211,6 +212,8 @@ public int read(long position, byte[] buffer, int offset, int length)
     if (streamStatistics != null) {
       streamStatistics.readOperationStarted();
     }
+    LOG.debug("Direct read with position called, no Optimizations");
+    tracingContext.setReaderID("NR");
```

Comment:
we should have a new tracingContext object in the method, else parallel thread would use same tc and probably metric-corruption can happen.